docs/source/architecture/transactions_and_batches.rst

```diff
@@ -0,0 +1,189 @@
+*************************
+Transactions and Batches!
+*************************
+
+Modifications to state are performed by creating and applying transactions.  A
+client creates a transaction and submits it to the validator.  The validator
+applies the transaction which causes a change to state.
+
+Transactions are always wrapped inside of a batch.  All transactions within a
+batch are committed to state together or not at all.  Thus, batches are the
+atomic unit of state change.
+
+The overall structure of batches and transactions includes Batch, BatchHeader,
+Transaction, and TransactionHeader:
+
+.. image:: ../images/arch_batch_and_transaction.*
+   :width: 80%
+   :align: center
+   :alt: Transaction and batch entity diagram
+
+Transaction Data Structure
+==========================
+
+Transactions are serialized using Protocol Buffers.  They consists of two
+message types:
+
+.. literalinclude:: ../../../protos/transaction.proto
+   :language: protobuf
+   :caption: File: protos/transaction.proto
+   :linenos:
+
+Header, Signature, and Public Keys
+----------------------------------
+
+The Transaction header field is a serialized version of a TransactionHeader.
+The header is signed by the signer's private key (not sent with the
+transaction) and the resulting signature is stored in header_signature.  The
+header is present in the serialized form so that the exact bytes can be
+verified against the signature upon receipt of the Transaction.
+
+The verification process verifies that the key in signer_pubkey signed the
+header bytes resulting in header_signature.
+
+The batcher_pubkey field must match the public key used to sign the batch in
+which this transaction is contained.
+
+
+.. note::
+
+   The original header bytes as constructed from the sender are used
+   for verification of the signature.  It is not considered good practice to
+   de-serialize the header (for example, to a Python object) and then
+   re-serialize the header with the intent to produce the same byte sequence as
+   the original.  Serialization can be sensitive to programming language or
+   library, and any deviation would produce a sequence that would not match the
+   signature; thus, best practice is to alway use the original header bytes for
+   verification.
+
+Transaction Family
+------------------
+
+In Sawtooth Lake, the set of possible transactions are defined by an extensible
+system called transaction families.  Defining and implementing a new
+transaction family adds to the taxonomy of available transactions which can be
+applied. For example, in the :ref:`transaction-family-tutorial`, we
+define a transaction family called "xo" which defines a set of transactions for
+playing tic-tac-toe.
+
+In addition to the name of the transaction family (family_name), each
+transaction specifies a family version string (family_version).  The
```

Comment:
The feature that is made possible by the family_version (making updates to a txn family and coordinating the nodes in the network to upgrade) seems like a unique feature that isn't obvious from just the mention that there is a family_version string. This should be called out either here or elsewhere.